### PR TITLE
fix: Fix common::grpc compiler errors

### PR DESCRIPTION
**Pull request by evenyag**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?
This commit refactors RecordBatch and holds vectors in the RecordBatch struct, so we don't need to cast the array to vector when doing serialization or iterating the batch.

Now we use the vector API instead of the arrow API in the `common-grpc` crate.

## Checklist

- [x]  I have written the necessary rustdoc comments.
- [x]  I have added the necessary unit tests and integration tests.

## Refer to a related PR or issue link (optional)
- #555

### CODE REVIEW

File: src/common/grpc/src/select.rs [hunk 1]
```diff
@@ -147,52 +152,114 @@ macro_rules! convert_arrow_array_to_grpc_vals {
...
-                            .iter()
+                            .iter_data()
                             .filter_map(|i| i.map($MapFunction))
                             .collect::<Vec<_>>());
                     }
                     return Ok(vals);
                 },
             )+
-            _ => unimplemented!(),
+            ConcreteDataType::Null(_) | ConcreteDataType::List(_) => unreachable!("Should not send {:?} in GRPC", $data_type),
```
Comment (waynexia) [inline src/common/grpc/src/select.rs:162]:
```suggestion
            ConcreteDataType::Null(_) | ConcreteDataType::List(_) => unreachable!("Should not send {:?} in gRPC", $data_type),
```

File: src/common/grpc/src/select.rs [hunk 2]
```diff
@@ -147,52 +152,114 @@ macro_rules! convert_arrow_array_to_grpc_vals {
...
-        (DataType::LargeBinary,   BinaryArray,            binary_values,  |x| {x.into()}),
-
-        (DataType::Utf8,          StringArray,            string_values,  |x| {x.into()}),
-        (DataType::LargeUtf8,     StringArray,            string_values,  |x| {x.into()}),
-
-        (DataType::Date32,        PrimitiveArray<i32>,    date_values,    |x| {*x as i32}),
-        (DataType::Date64,        PrimitiveArray<i64>,    datetime_values,|x| {*x as i64}),
-
-        (DataType::Timestamp(TimeUnit::Millisecond, _), PrimitiveArray<i64>, ts_millis_values, |x| {*x})
+        data_type,
```
Comment (waynexia) [inline src/common/grpc/src/select.rs:174]: The previous unformat version is more clear. Can you revert this change?

File: src/script/Cargo.toml [hunk 3]
```diff
@@ -43,18 +43,18 @@ paste = { version = "1.0", optional = true }
...
 rustpython-ast = { git = "https://github.com/discord9/RustPython", optional = true, rev = "183e8dab" }
+rustpython-codegen = { git = "https://github.com/discord9/RustPython", optional = true, rev = "183e8dab" }
 rustpython-compiler = { git = "https://github.com/discord9/RustPython", optional = true, rev = "183e8dab" }
 rustpython-compiler-core = { git = "https://github.com/discord9/RustPython", optional = true, rev = "183e8dab" }
-rustpython-codegen = { git = "https://github.com/discord9/RustPython", optional = true, rev = "183e8dab" }
 rustpython-parser = { git = "https://github.com/discord9/RustPython", optional = true, rev = "183e8dab" }
+rustpython-pylib = { git = "https://github.com/discord9/RustPython", optional = true, rev = "183e8dab", features = [
+    "freeze-stdlib",
+] }
+rustpython-stdlib = { git = "https://github.com/discord9/RustPython", optional = true, rev = "183e8dab" }
```
Comment (waynexia) [inline src/script/Cargo.toml:53]: Do this stuff in `dev` and merge it into `replace-arrow2`. I'm afraid it will conflict when merging `replace-arrow2`.

File: src/common/grpc/src/select.rs (hunk 2 above)
Comment (evenyag) [reply src/common/grpc/src/select.rs:174]: Unfortunately, this is formatted by `cargo fmt` and could not be reverted.

**Review by v0y4g3r — APPROVED** [review]
LGTM

**Review by waynexia — APPROVED** [review]
